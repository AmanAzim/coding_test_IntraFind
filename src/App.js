import React, {useState} from 'react';

function App() {
  const [userData, setUserData] = useState([]);
  const [fname, setFname] = useState('');
  const [lname, setLname] = useState('');
  const [email, setEmail] = useState('');
  const [editFlag, setEditFlag] = useState({editing: false, id: undefined});

  const handleChange = (event) => {
     if (event.target.name === 'fname') {
        setFname(event.target.value);
     }
     if (event.target.name === 'lname') {
         setLname(event.target.value);
     }
     if (event.target.name === 'email') {
         setEmail(event.target.value);
     }
  };

  const onSubmit = (event) => {
      event.preventDefault();
      if (!editFlag.editing) {
        setUserData([...userData, {fname, lname, email}]);
        setFname('');
        setLname('');
        setEmail('');
        //Make http request to backend to save changed data e.g. axios.post()...
      }
  };

  const onDelete = (id) => {
     const changedData = userData.filter((data, idx) => idx!==id);
     setUserData(changedData);
  };

  const onEdit = (id) => {
     const dataToEdit = userData.filter((data, idx) => idx===id);
     setFname(dataToEdit[0].fname);
     setLname(dataToEdit[0].lname);
     setEmail(dataToEdit[0].email);
     setEditFlag({editing: true, id});
  };

  const onSave = () => {
     const editedData = userData.map(data=>data);// quick solution for copying reference type
     editedData[editFlag.id].fname = fname;
     editedData[editFlag.id].lname = lname;
     editedData[editFlag.id].email = email;
     setUserData(editedData);
     setFname('');
     setLname('');
     setEmail('');
     setEditFlag({editing: false, id: undefined});
     //Make http request to backend to save changed data e.g. axios.post()...
  };

  return (
    <div style={{margin: '30px'}}>
       <form onSubmit={onSubmit}>
          First name: <input type="text" name="fname" value={fname} onChange={handleChange}/><br/>
          Last name: <input type="text" name="lname" value={lname} onChange={handleChange}/><br/>
          Email: <input type="email" name="email" value={email} onChange={handleChange}/><br/>
           {!editFlag.editing && <input type="submit" value="Submit" />}
           {editFlag.editing  && <button onClick={onSave}>Save</button>}
        </form>
        <br/><br/>
        <hr />
        <hr />
        <b>Entered user data:</b><br/><br/>
        {userData.map( (data, id) => {
           return (
             <div key={id}>
                First Name: {data.fname}<br/>
                Last Name: {data.lname}<br/>
                Email: {data.email}<br/>
                <button onClick={()=>onEdit(id)}>Edit</button>
                <button onClick={()=>onDelete(id)}>Delete</button>
                <br/><br/>
             </div>
           );
        })}
    </div>
  );
};

export default App;
