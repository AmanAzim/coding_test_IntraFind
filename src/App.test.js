import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('App.js renders properly', () => {
    const wrappper = renderer.create(<App />).toJSON();
    expect(wrappper).toMatchSnapshot();
});
