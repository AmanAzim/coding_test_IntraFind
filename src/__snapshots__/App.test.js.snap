// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js renders properly 1`] = `
<div
  style={
    Object {
      "margin": "30px",
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    First name: 
    <input
      name="fname"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    Last name: 
    <input
      name="lname"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    Email: 
    <input
      name="email"
      onChange={[Function]}
      type="email"
      value=""
    />
    <br />
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <br />
  <br />
  <hr />
  <hr />
  <b>
    Entered user data:
  </b>
  <br />
  <br />
</div>
`;
